src/lib/config/config.mjs

```diff
@@ -68,36 +90,68 @@ function validateConfig(profileName, profileBody, configPath) {
   }
 }
 
-export function configParser(path) {
-  let parsedProfile;
-  const logger = container.resolve("logger");
+/**
+ * Creates a yargs parser given existing command line arguments
+ *
+ * @param {string|string[]} argvInput - The raw command line arguments.
+ * @returns {(path: string) => object} - The yargs parser
+ */
+export const configParser = (argvInput) =>
+  /**
+   * A parser to convert config files into appropriate command line arguments
+   *
+   * @param {string} path
+   * @returns {object}
+   */
+  function configParserWithArgs(path) {
+    let parsedPath = path;
+    let parsedProfile;
+    const logger = container.resolve("logger");
```

Comment:
this looks more complicated than it needs to be - want to pair on it today? i think we can get rid of the inner function and shake about half the lines of code.